sycl/test/regression/device_num.cpp

```diff
@@ -0,0 +1,94 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: sycl-ls
+// RUN: env SYCL_DEVICE_FILTER=0 %t.out
+// RUN: env SYCL_DEVICE_FILTER=1 %t.out
+// RUN: env SYCL_DEVICE_FILTER=2 %t.out
+// RUN: env SYCL_DEVICE_FILTER=3 %t.out
+// RUN: env SYCL_DEVICE_FILTER=4 %t.out
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+using namespace std;
+
+void printDeviceType(const device &d) {
+  string name = d.get_platform().get_info<info::platform::name>();
+  auto DeviceType = d.get_info<info::device::device_type>();
+  std::string DeviceTypeName;
+
+  switch (DeviceType) {
+  case info::device_type::cpu:
+    DeviceTypeName = "CPU ";
+    break;
+  case info::device_type::gpu:
+    DeviceTypeName = "GPU ";
+    break;
+  case info::device_type::host:
+    DeviceTypeName = "HOST ";
+    break;
+  case info::device_type::accelerator:
+    DeviceTypeName = "ACCELERATOR ";
+    break;
+  default:
+    DeviceTypeName = "UNKNOWN ";
+    break;
+  }
+  std::cout << DeviceTypeName << name << std::endl;
+}
+
+int main() {
+  const char *envVal = std::getenv("SYCL_DEVICE_FILTER");
+  int deviceNum;
+  std::cout << "SYCL_DEVICE_FILTER=" << envVal << std::endl;
+  deviceNum = *envVal - '0';
```

Comment:
It is unlikely that someone would have 10+ devices, but still possible.
please use atoi:
```suggestion
  deviceNum = std::atoi(envVal);
```